solr/core/src/java/org/apache/solr/servlet/PrometheusMetricsServlet.java

```diff
@@ -741,11 +747,12 @@ private static String buildProperty() {
     }
 
     @Override
-    protected void handle(List<PrometheusMetric> results, JsonNode metricsNode) throws IOException {
-      missingCoreMetrics.clear();
+    protected void handle(ResultContext resultContext, JsonNode metricsNode) throws IOException {
+      List<PrometheusMetric> results = resultContext.resultMetrics;
       JsonNode nodeMetricNode = metricsNode.get("solr.node");
 
       if (nodeMetricNode != null) {
+        resultContext.missingCoreMetrics = new ArrayList<>(); // explicitly set missing core metrics
         for (CoreMetric metric : CoreMetric.values()) {
```

Comment:
This is the usual case, right? Assuming so, why not just directly set `final missingCoreMetrics = new ArrayList<>()` on the `ResultContext` object when it's constructed? Then we don't need this lazy initialization.

Though it looks like lazy init here (allowing `missingCoreMetrics==null` is interpreted by the downstream metrics caller to just pull all metrics. That seems fine actually, so nevermind the above question.

Reply:
yep answered in another comment :) . there's indeed a difference between null and empty list